tutorial/sample-problems/Misc/Matching.pg

```diff
@@ -0,0 +1,126 @@
+## DESCRIPTION
+## Matching question template
+## ENDDESCRIPTION
+
+## KEYWORDS('matching', 'template')
+
+## DBsubject(WeBWorK)
+## DBchapter(WeBWorK tutorial)
+## DBsection(PGML tutorial 2015)
+## Date(06/01/2015)
+## Institution(Hope College)
+## Author(Paul Pearson)
+
+#:% name = Matching Problem
+#:% type = Sample
+#:% categories = [multiple choice, misc]
+
+#:% section = preamble
+#: The `PGchoicemacros.pl` macro is used to construct the list of multiple
+#: choice items, and the custom problem grader fluid from `PGgraders.pl` is
+#: used for incremental grading.
+DOCUMENT();
+
+loadMacros(
+    'PGstandard.pl',     'PGML.pl',
+    'PGchoicemacros.pl', 'PGgraders.pl',
+    'unionTables.pl',    'PGcourse.pl'
+);
+
+#:% section = setup
+#: Withhold feedback when answers are submitted by setting
+#: `$showPartialCorrectAnswers = 0;`.
+#:
+#: This problem uses an incremental grader called the `custom_problem_grader_fluid`.
+#: With this problem grader, the number of correct answers `[2, 4, 6]`
+#: and their corresponding scores `[0.3, 0.6, 1]` must be specified.  The last
+#: entry in the `grader_numright` array must be the total number of questions
+#: asked, and the last entry in the `grader_scores` array must be 1
+#: (otherwise nobody can earn full credit!).  The grader message can also
+#: be customized by setting the value of `grader_message` to the desired custom
+#: message.
+#:
+#: If a grader is desired that awards full credit when all questions are correct
+#: and no credit otherwise, use the commented out standard problem grader code
+#: instead.
+#:
+#: Create a list of 6 questions and answers, 2 extra answers, and a
+#: 'None of the above' answer that will be made last with `makeLast`.
+#: So the popup list must have 9 entries A through I.
+#:
+#: As an alternative, see PROBLINK('MatchingAlt.pl') for another way to write
```

Comment:
The `PROBLINK` call should be `PROBLINK('MatchingAlt.pg')`.  When you run the `parser-problem-doc.pl` script or view the problem in webwork2 it gives a "use of uninitialized value in concatentation (.) or string" warning from this.